### PR TITLE
Fix Char Swap crash (again) and other input bugs

### PR DESCRIPTION
**Pull request by warblgarbl**
Fixed crash when selecting an empty slot with keyboard/controller.
Fixed mouse click highlight on locked party members.

To account for disabling unlock party with a non-retail party configuration, the party is checked for certain conditions when the character swap screen is loaded. If any of the following are true, the party will be reorganized:
* Swappable character is in a higher slot than a required character
* Slot is empty
* Required character is in the secondary/reserve roster
* First slot is not a required character

~Note:  The only non-retail outcome that can still occur is having required characters in slot 0 and slot 2, with a swappable slot 1.~
_Edit: Locked characters now sort to top. Example of remaining exceptions:
Hellena Prison 2
0 Lavitz
1 Dart
2 Rose_

### CODE REVIEW

File: src/main/java/legend/game/inventory/screens/CharSwapScreen.java [hunk 1]
```diff
@@ -49,6 +49,57 @@ protected void render() {
         this.primaryCharIndex = 0;
         this.secondaryCharIndex = 0;
 
+        if(!CONFIG.getConfig(CoreMod.UNLOCK_PARTY_CONFIG.get())) {
```
Comment (LordMonoxide) [inline src/main/java/legend/game/inventory/screens/CharSwapScreen.java:52]: Can you sprinkle some comments in here to explain what these various loops and stuff are doing?

File: src/main/java/legend/game/inventory/screens/CharSwapScreen.java [hunk 2]
```diff
@@ -49,6 +49,57 @@ protected void render() {
...
+          boolean sortPrimary = false;
+          boolean requiredInSecondary = false;
+          int primarySlotIndex;
+          int secondarySlotIndex;
+
+          for(primarySlotIndex = 2; primarySlotIndex > 0 && !sortPrimary; primarySlotIndex--) {
```
Comment (warblgarbl) [inline src/main/java/legend/game/inventory/screens/CharSwapScreen.java:58]: Checking for empty slots or swappable characters placed above locked characters

File: src/main/java/legend/game/inventory/screens/CharSwapScreen.java [hunk 3]
```diff
@@ -49,6 +49,57 @@ protected void render() {
...
+            final int charA = gameState_800babc8.charIds_88[primarySlotIndex];
+            final int charB = gameState_800babc8.charIds_88[primarySlotIndex - 1];
+            sortPrimary = charA == -1 || charB == -1 || ((gameState_800babc8.charData_32c[charA].partyFlags_04 & 0x20) != 0 && (gameState_800babc8.charData_32c[charB].partyFlags_04 & 0x20) == 0);
+          }
+          for(secondarySlotIndex = 0; secondarySlotIndex < 6 && !sortPrimary && !requiredInSecondary; secondarySlotIndex++) {
```
Comment (warblgarbl) [inline src/main/java/legend/game/inventory/screens/CharSwapScreen.java:63]: Checking for locked characters not in the active party

File: src/main/java/legend/game/inventory/screens/CharSwapScreen.java [hunk 4]
```diff
@@ -49,6 +49,57 @@ protected void render() {
...
+              charIndex = gameState_800babc8.charIds_88[i];
+              if(charIndex != -1 && (gameState_800babc8.charData_32c[charIndex].partyFlags_04 & 0x2) != 0 && charIndex != slots[0] && charIndex != slots[1] && charIndex != slots[2]) {
+                slots[primarySlotIndex++] = charIndex;
+              }
+            }
+            for(charIndex = 0; charIndex < 9 && primarySlotIndex < 3; charIndex++) {
+              if((gameState_800babc8.charData_32c[charIndex].partyFlags_04 & 0x2) != 0 && charIndex != slots[0] && charIndex != slots[1] && charIndex != slots[2]) {
+                slots[primarySlotIndex++] = charIndex;
+              }
+            }
```
Comment (warblgarbl) [inline src/main/java/legend/game/inventory/screens/CharSwapScreen.java:86]: Building new party. Priority is per loop.
1. Locked/required characters.
2. Available characters in current party.
3. Available characters not in current party.

File: src/main/java/legend/game/inventory/screens/CharSwapScreen.java [hunk 5]
```diff
@@ -49,6 +49,57 @@ protected void render() {
...
+          int secondarySlotIndex;
+
+          for(primarySlotIndex = 2; primarySlotIndex > 0 && !sortPrimary; primarySlotIndex--) {
+            final int charA = gameState_800babc8.charIds_88[primarySlotIndex];
+            final int charB = gameState_800babc8.charIds_88[primarySlotIndex - 1];
+            sortPrimary = charA == -1 || charB == -1 || ((gameState_800babc8.charData_32c[charA].partyFlags_04 & 0x20) != 0 && (gameState_800babc8.charData_32c[charB].partyFlags_04 & 0x20) == 0);
+          }
+          for(secondarySlotIndex = 0; secondarySlotIndex < 6 && !sortPrimary && !requiredInSecondary; secondarySlotIndex++) {
+            requiredInSecondary = secondaryCharIds_800bdbf8[secondarySlotIndex] != -1 && (gameState_800babc8.charData_32c[secondaryCharIds_800bdbf8[secondarySlotIndex]].partyFlags_04 & 0x20) != 0;
+          }
```
Comment (warblgarbl) [inline src/main/java/legend/game/inventory/screens/CharSwapScreen.java:65]: Checking for locked characters not in the active party

**Review by LordMonoxide — APPROVED** [review]

**Pull request merged**